### ISSUE TITLE
Feature request - readonly ref returns

### ISSUE THREAD

**Issue by jayrulez** [beefytech/Beef#1412]
Read-only ref returns is the counterpart to in and is useful when you want to return a reference but do not intend it to be modifiable.

Relevant reading is here:
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-7.2/readonly-ref

**Comment by bfiete** [comment]
Actually, we already support this, except for us it's `readonly ref` instead of `ref readonly`, which is consistent with the fact that _any_ local variable can be `readonly`.

```C#
static int sA = 123;

static readonly ref int GetRef()
{
	return ref sA;
}

public static int Main(String[] args)
{
	readonly String str = scope .();
	readonly ref int a = ref GetRef();

	str.Append("Abc"); // Works
	str = null; // ERROR: Not allowed due to readonly
	a = 123; // ERROR: Not allowed due to readonly

	return 0;
}
```

**Comment by jayrulez** [comment]
Well there is an issue where the readonly is removed if you use Ctrl+K+D to format the code.

**Comment by bfiete** [comment]
Fixed at 4c10fc0e423aff7c5d4eb2e44ad0a7bb939c1372

**Comment by bfiete** [comment]
Fails to format for properties https://discord.com/channels/670260150656827422/703607007277744150/950097185423360000

**Comment by bfiete** [comment]
Fixed at 7dc09b38f9b9bf4600e552290155bb642332d7db
